src/Firebase/Auth/ApiClient.php

```diff
@@ -183,6 +184,62 @@ public function unlinkProvider(string $uid, array $providers): ResponseInterface
         ]);
     }
 
+    /**
+     * @param array<array<string|int>> $users The users to import
+     * @param array<array<string|int>> $options Import options. Used to specify how passwords are hashed for example
+     *
+     * @throws AuthException
+     * @throws FirebaseException
+     */
+    public function importUsers(array $users, array $options, ProjectId $projectId): ResponseInterface
+    {
+        // Determine the request body
+        $body = [];
+        if (isset($options['hash'])) {
+            if (isset($options['hash']['algorithm'])) {
+                $body['hashAlgorithm'] = $options['hash']['algorithm'];
+            }
+            if (isset($options['hash']['rounds'])) {
+                $body['rounds'] = $options['hash']['rounds'];
+            }
+            if (isset($options['hash']['key'])) {
+                $body['signerKey'] = $options['hash']['key'];
+            } // Base64-encoded
+            if (isset($options['hash']['memoryCost'])) {
+                $body['cpuMemCost'] = $options['hash']['memoryCost'];
+            }
+            if (isset($options['hash']['parallelization'])) {
+                $body['parallelization'] = $options['hash']['parallelization'];
+            }
+            if (isset($options['hash']['blockSize'])) {
+                $body['blockSize'] = $options['hash']['blockSize'];
+            }
+            if (isset($options['hash']['derivedKeyLength'])) {
+                $body['dkLen'] = $options['hash']['derivedKeyLength'];
+            }
+            if (isset($options['hash']['saltSeparator'])) {
+                $body['saltSeparator'] = $options['hash']['saltSeparator'];
+            } // Base64-encoded
+        }
+        $body['users'] = \array_map(static function (array $userData) {
+            $res = \array_merge([
+                'localId' => $userData['uid'],
+            ], $userData);
+            unset($res['uid']);
+            // Make sure to pass the hashes in a base64-encoded format
+            if (isset($res['passwordSalt'])) {
+                $res['salt'] = $userData['passwordSalt'];
+                unset($res['passwordSalt']);
+            }
+
+            return $res;
+        }, $users);
```

Comment:
I would change this to:

```php
$body['users'] = [];

foreach ($users as $userData) {
    $userData['localId'] = $userData['uid'] ?? null;
    $userData['salt'] = $userData['passwordSalt'] ?? null;
    unset($userData['uid'], $userData['passwordSalt']);

    $body['users'][] = $userData;
}
```

Reply:
Can we import the users without passwords using this? I have tried to send Facebook provider details like below, but it is not working. How to import existing Facebook member using this function?.

$body['users'] = [[
            localId => 'c9fe1fdb715d05c4c4758e86286',            
            displayName => 'Siva Raj',
            email => 'siva_mpaiefz_raj@tfbnw.net',
            photoURL => 'http://www.example.com/12345678/photo.png',
            emailVerified => true,
            phoneNumber => '+11234567890',
            providerData => [[
              uid => '104653111407187',
              email => 'siva_mpaiefz_raj@tfbnw.net',
              displayName => 'Siva Raj',
              photoURL => 'http://www.example.com/12345678/photo.png',
              providerId => 'facebook.com'
            ]]
        ]];